### PR TITLE
adding hparams as parameters during model initialization

### PR DESCRIPTION
**Pull request by nthain**
WARNING: This is not backwards compatible with older model versions as it requires the model hyperparameters to be saved and loaded alongside the model.

### CODE REVIEW

File: src/model_tool.py [hunk 1]
```diff
@@ -51,25 +50,48 @@ def compute_auc(y_true, y_pred):
...
+        self.hparams.update(new_hparams)
         
     def get_model_name(self):
         return self.model_name
 
+    def save_hparams(self, model_name):
+        self.hparams['model_name'] = model_name
+        cPickle.dump(self.hparams, 
+            open(os.path.join(self.model_dir, 
+                '%s_hparams.pkl' % self.model_name), 'wb'))
```
Comment (jetpack) [inline src/model_tool.py]: I think pickle is generally discouraged: http://www.benfrederickson.com/dont-pickle-your-data/

This data is pretty simple. Can we save it as JSON instead? Also makes it nicely human readable.

File: src/model_tool.py [hunk 2]
```diff
@@ -91,12 +113,13 @@ def prep_data(self, data_path, text_column, label_column = None,
         data = pd.read_csv(data_path)
```
Comment (jetpack) [inline src/model_tool.py]: Remove max_sequence_length from docstring.

File: src/model_tool.py (hunk 1 above)
Comment (nthain) [reply src/model_tool.py]: Done.

File: src/model_tool.py (hunk 2 above)
Comment (nthain) [reply src/model_tool.py]: Done.

**Comment by nthain** [comment]
Resolved comments. Also added functionality to save only the best model that is being trained. PTAL.

**Pull request merged**